parquet-column/src/main/java/org/apache/parquet/column/impl/ColumnReadStoreImpl.java

```diff
@@ -72,7 +72,13 @@ public ColumnReadStoreImpl(PageReadStore pageReadStore,
 
   @Override
   public ColumnReader getColumnReader(ColumnDescriptor path) {
-    return newMemColumnReader(path, pageReadStore.getPageReader(path));
+    PrimitiveConverter converter = getPrimitiveConverter(path);
+    PageReader pageReader = pageReadStore.getPageReader(path);
+    if (pageReadStore.isInPageFilteringMode()) {
+      return new SynchronizingColumnReader(path, pageReader, converter, writerVersion, pageReadStore.getRowIndexes());
+    } else {
+      return new ColumnReaderImpl(path, pageReader, converter, writerVersion);
```

Comment:
Why doesn't this use `newMemColumnReader`? Since there are only two uses of that function, I think that either both of them should be inlined like this, or both should continue calling it.

Reply:
`newMemColumnReader` is used by `ParquetFileWriter.merge(List<InputFile>, BytesCompressor, String, long)` introduced in PARQUET-1381. It is different from `getColumnReader(ColumnDescriptor)` as it gets the `PageReader` instead of retrieving it from the store.